store/store.go

```diff
@@ -914,6 +932,7 @@ func (s *Store) Find(search *Search, user *auth.UserState) ([]*snap.Info, error)
 	} else {
 		q.Set("q", searchTerm)
 	}
+	q.Set("section", search.Section)
```

Comment:
Its "funny". This is actually required for the store to work currently. It behaves differently for `section=` than for a missing section in the query:
```
$ curl -s -H "accept: application/hal+json" -H "X-Ubuntu-Release: 16" -H "X-Ubuntu-Device-Channel: stable" -H "X-Ubuntu-Wire-Protocol: 1" -H "X-Ubuntu-Architecture: amd64"  'https://search.apps.ubuntu.com/api/v1/snaps/search?section=' |python3 -m json.tool|grep '"title"'
                "title": "Docker",
                "title": "lxd",
                "title": "mongo32",
                "title": "Rocket Chat Server",
```
However:
```
$ curl -s -H "accept: application/hal+json" -H "X-Ubuntu-Release: 16" -H "X-Ubuntu-Device-Channel: stable" -H "X-Ubuntu-Wire-Protocol: 1" -H "X-Ubuntu-Architecture: amd64"  'https://search.apps.ubuntu.com/api/v1/snaps/search?q=' |python3 -m json.tool|grep '"title"'|wc -l
100
```
i.e. the later query will just return everything (paginated).